lib/msal-common/src/utils/FunctionWrappers.ts

```diff
@@ -53,9 +57,8 @@ export const invoke = <T extends Array<any>, U>(
             } catch (e) {
                 logger.trace("Unable to print error message.");
             }
-            inProgressEvent?.end({
-                success: false,
-            });
+
+            captureError(e, inProgressEvent);
```

Comment:
This would capture every error that occurs in the stack, even ones we end up swallowing, potentially leading to error fields being defined in successful calls. It may be better to handle this at the top of the stack only, for example: [here](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dfda8fa0751a1dc924d081ab359a25860b22483f/lib/msal-browser/src/controllers/StandardController.ts#L743)

Reply:
That's a very good point. Thanks. I have made the change such that we capture errors in controllers and simplified the instrumentation code.